build/teamcity-roachtest-stress.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+source "$(dirname "${0}")/teamcity-support.sh"
+
+google_credentials="$GOOGLE_EPHEMERAL_CREDENTIALS"
+generate_ssh_key
+log_into_gcloud
+export ROACHPROD_USER=teamcity
+
+build/builder/mkrelease.sh amd64-linux-gnu build bin/workload bin/roachtest bin/roachprod 
```

Comment:
How do you feel about making this happen for the main nightly jobs as well? We currently depend on publish bleeding edge but that's just a bad idea. I know it's a separate topic but we lose a roachtest run or two every week on that dependency, so while you're poking around here it might be a good idea to get rid of it.

There is something to be said about testing *exactly* the released binary, but I don't know, using the exact same build process ought to be enough for now given that dev-inf is not likely to disentangle the dependency for us anytime soon.

Reply:
I suppose that should work, assuming there are no implicit dependencies on what that build does (which is a lot). I'd feel better about cleaning this up properly, or even just adding a minimal new canonical build that we could depend on here and expand for other builds, but as a temporary bandaid it should be fine.

Reply:
There are no implicit dependencies. If there are, I sure wouldn't mind knowing about them! I think we should move the main job to build its own but it doesn't have to be in this PR.

Reply:
Added a separate commit for this, we'll need to change the build once this is merged (possibly after a nightly run to make sure it works).

Reply:
Want to give the line this comment is anchored on the same treatment as in the main script (redirect output, hide if successful)?

Reply:
Sure, done.